src/functions/src/main/java/org/apache/jmeter/functions/DigestEncodeFunction.java

```diff
@@ -103,12 +100,9 @@ public String execute(SampleResult previousResult, Sampler currentSampler) throw
      * @return
      */
     private String uppercase(String encodedString, CompoundVariable[] values, int index) {
-        if (values.length > index) {
-            String shouldUpperCase = values[index].execute();
-            boolean shouldDoUpperCase = Boolean.TRUE.toString().equalsIgnoreCase(shouldUpperCase);
-            if (shouldDoUpperCase) {
-                return encodedString.toUpperCase();
-            }
+        String shouldUpperCase = values.length > index ? values[index].execute() : null;
```

Comment:
Do you think `"false"` be better than `null` here?

Reply:
@ham1 parseBoolean have null check, so it isn't a must
And I don't want to add hardcoded default value

Reply:
Just in case, I agree with you both. However, this PR is probably attracting people.
They open, browse, and leave :(

I guess we should have a faster turnaround so we don't spend time on trivial improvements.

Should it be

```java
String shouldUpperCase = values.length < index && Boolean.parseBoolean(values[index].execute());
```
?

or even

```
if (values.length >= index ||  Boolean.parseBoolean(values[index].execute())) {
    return;
} 
```

?

Should there be a dedicated `Hex.encodeHex` implementation that converted to the proper casing, so we don't need to throw-away "lowercase" string?


Well, `Hex.encodeHex` is kind out of the scope, however, it might be a bit more interesting question to discuss rather than the way we write boolean expressions :-/

Reply:
@vlsi Remove `.name()`, notice there's a new enhnacement for Hex function, so it can be further discuss Hex ecnoding there